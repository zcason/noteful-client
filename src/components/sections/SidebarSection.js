import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import FolderItem from '../FolderItem';
import DefaultContext from '../context/DefaultContext';
import PropTypes from 'prop-types';

class SidebarSection extends Component {
    static contextType = DefaultContext;
    getFolderList = () => {
        if ( typeof this.props.store !== "object" || this.props.history.location.pathname.includes("/note/") ) return;
        return this.props.store.folders.map(folder => {
            return (
                <FolderItem
                    key={folder.id}
                    id={folder.id}
                    name={folder.name}
                    store={this.props.store}
                />
            );
        });
    }
    getBottomButton = () => {
        if ( !this.props.history.location.pathname.includes("/note/") ) {
            return (
            <Link 
                to={`/add-folder/`}
                className="folder--item--add"
            >
                Add Folder
            </Link>);
        }

        return (<Link 
                    to={`/`}
                    className="button--back"
                >
                    Go back
                </Link>
        );
    }
    getFolderName = () => {
        if (this.props.history.location.pathname.includes("/note/")) {
            const folder = this.props.store.folders.find( folder => {
                return folder.id === this.props.store.notes[0].folderId
            } );
            return (
                <h3>Folder: {folder.name}</h3>
            )
        }
    }
    render(){
        return (
            <section className="section--sidebar">
                <ul>
                    {this.getFolderList()}
                </ul>
                {this.getBottomButton()}
                {this.getFolderName()}
            </section>
       );
    }
}

SidebarSection.propTypes = {
    store: PropTypes.object.isRequired
}

export default withRouter(SidebarSection);
