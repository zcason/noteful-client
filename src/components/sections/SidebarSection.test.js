import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import store from '../../dummy-store';
import SidebarSection from './SidebarSection';

describe(`SidebarSection component`, () => {
    const props = {
      store: store
    }
  
    it('renders SidebarSection by default', () => {
      const wrapper = shallow(<SidebarSection />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders Sidebar section with store when in props', () => {
      const component = shallow(<SidebarSection {...props} />)
        .find('.section--sidebar')
      expect(toJson(component)).toMatchSnapshot()
    })
  })
  