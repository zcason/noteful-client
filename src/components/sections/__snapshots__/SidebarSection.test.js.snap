// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarSection component renders Sidebar section with store when in props 1`] = `null`;

exports[`SidebarSection component renders SidebarSection by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
